### DISCUSSION TITLE
How does it work with other languages?

### DISCUSSION THREAD

**Discussion by SiriusFang2020** [exaloop/codon#120] [Q&A]
How does it work with other languages?
For example, like java calls up C++ with JNA/JNI?

**Comment by arshajii** [comment]
Calling C functions is actually very easy since we compile to LLVM IR anyway. We also support calling plain Python functions/libraries; see https://docs.exaloop.io/codon/language/ffi for some more info (essentially Codon does all the object conversions and CPython API calls under the hood). It's possible we'll add more languages (e.g. R) in the future if there's demand for that!
